app/models/team.rb

```diff
@@ -184,11 +185,21 @@ def disallow_duplicate_members
   end
 
   def limit_number_of_students
-    students = roles.select{|r| r.name == 'student' && !r.marked_for_destruction?}
+    students = members_with_role 'student'
     return unless students.size > 2
     errors.add(:roles, 'there cannot be more than 2 students on a team.')
   end
 
+  def limit_number_of_coaches
+    coaches = members_with_role 'coach'
+    return unless coaches.size > 4
```

Comment:
Can you pull that on one line, eliminating the need for the local variable?

Reply:
Done! Thanks for the suggestion